internal/eval/evalers.go

```diff
@@ -1073,3 +1120,123 @@ func (n *ipIsInRangeEval) Eval(ctx *Context) (types.Value, error) {
 	}
 	return types.Boolean(rhs.Contains(lhs)), nil
 }
+
+// virtualLessThanEval struct
+type virtualLessThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualLessThanEval(lhs Evaler, rhs Evaler) *virtualLessThanEval {
+	return &virtualLessThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualLessThanEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(lhs.Less(rhs)), nil
+}
+
+var _ Evaler = &virtualLessThanEval{}
+
+// virtualGreaterThanEval struct
+type virtualGreaterThanEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualGreaterThanEval(lhs Evaler, rhs Evaler) *virtualGreaterThanEval {
+	return &virtualGreaterThanEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualGreaterThanEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(!lhs.LessEqual(rhs)), nil
+}
+
+// virtualLessEqualThanEval struct
+type virtualLessThanOrEqualEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualLessThanOrEqualEval(lhs Evaler, rhs Evaler) *virtualLessThanOrEqualEval {
+	return &virtualLessThanOrEqualEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualLessThanOrEqualEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(lhs.LessEqual(rhs)), nil
+}
+
+// virtualGreaterEqualThanEval struct
+type virtualGreaterThanOrEqualEval struct {
+	lhs Evaler
+	rhs Evaler
+}
+
+func newVirtualGreaterThanOrEqualEval(lhs Evaler, rhs Evaler) *virtualGreaterThanOrEqualEval {
+	return &virtualGreaterThanOrEqualEval{
+		lhs: lhs,
+		rhs: rhs,
+	}
+}
+
+func (n *virtualGreaterThanOrEqualEval) Eval(ctx *Context) (types.Value, error) {
+	lhs, err := evalLesser(n.lhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	rhs, err := evalLesser(n.rhs, ctx)
+	if err != nil {
+		return zeroValue(), err
+	}
+	return types.Boolean(!lhs.Less(rhs)), nil
+}
```

Comment:
I think it might need to be an error if the types are not the same.  Maybe this is part of the Less signature `Less(rhs) (bool, error)`

Reply:
I agree that it should be an error. Happy to field suggestions for how to make that happen. I don't think a string comparison is it. Maybe the `Less` functions need to be `(bool, error)` since they actually can introspect values.

Reply:
And I just realized, hold this thought for a bit.  There's a line of work that needs to happen on our end.  (We might be adding comparison methods to all value types that work as-per go `slices.SortFunc` requires.

Reply:
There might even have to be two different comparisons, one for extension purposes, one for other purposes.

Reply:
I'll leave for now.

Reply:
One other (annoying) thought is that since the comparison error message depends on which is the LHS vs RHS we will either want two different methods (LessThan/MoreThan) or a single `Compare(rhs) int` so that we can be sure the type error complains about the RHS.

e.g. we want this 
```
5 < Duration("5s") // type error: expected long, got duration
5 > Duration("5s") // type error: expected long, got duration
```

not this
```
5 < Duration("5s") // type error: expected long, got duration
5 > Duration("5s") // type error: expected duration, got long
```